README.md

```diff
@@ -1,24 +1,161 @@
+
+```ruby
+
+# Post factory for reference:
+# title "My Great Post"
+# description "My amazing description"
+
+scenario 'title is shown on the show page in a h1 tag' do
+  visit "/posts/#{subject.id}"
+  expect(page).to have_css("h1", text: "My Great Post")
+end
+
+scenario 'description is shown on the show page in a p tag' do
+  skip
+end
+```
+
+This gives us a failure that says: ```expected to find css "h1" with text "My Great Post" but there were no matches```. To fix this we first need to capture the post's id in the controller's show action, let's do that here:
+
+```ruby
+# app/controllers/posts_controller.rb
+
+def show
+  @post = Post.find(params[:id])
+end
+```
+
+In this line, our show action is running a database query on the Post model that will return a post that has the id that matches the route parameters, and it will store this record in the ```@post``` instance variable and make it available to the ```show.html.erb``` file. Let's get our spec passing by placing the post's title on the show view template:
+
+```ERB
+<% # app/views/posts/show.html.erb %>
+<h1><%= @post.title %></h1>
+```
+
+We're back to all green! Now let's implement the description spec:
+
+```ruby
+scenario 'description is shown on the show page in a p tag' do
+  visit "/posts/#{subject.id}"
+  expect(page).to have_css("p", text: "My amazing description")
+end
+```
+This will give us a failure since there are no matches on the template yet, to implement this fix, update the view:
+
+```ERB
+<% # app/views/posts/show.html.erb %>
+<h1><%= @post.title %></h1>
+<p><%= @post.description %></p>
+```
+
+Now we're all passing, you now know how to create dynamic routes in Rails! However we would be remiss if we didn't follow the full "Red, Green, Refactor" TDD workflow. There are a few elements of the application that can be refactored. First let's add some syntactic sugar to our route file.
+
+Remove:
+
+```ruby
+get 'posts/:id', to: 'posts#show'
+```
+
+Replace with:
+
+```ruby
+resources :posts, only: :show
+```
+
+Running the tests again everything is still passing. Now let's update the specs. In each instance of:
+
+```ruby
+visit "/posts/#{subject.id}"
+```
+
+Replace that with:
+
+```ruby
+visit post_path(subject)
+```
+
+The tests are still passing and it's considered a better practice to leverage build in route methods as opposed to hard coding paths. Lastly, let's refactor the controller, our ```Post.find``` method works great, but what happens when we want to make this available to other methods in the controller? It would be better if we had a before action that will store the ```@post``` instance variable for the methods that should have it. The ```posts_controller.rb``` file should be refactored to look something like this:
```

Comment:
Also I would drop this refactor as well. They don't know what before_action does, they are probably quite confused as to all the moving parts already and this line of code doesn't look any better then the previous stuff. Before we show this, we want them to *feel the pain* that the `before_action` resolves. So just drop this entire section